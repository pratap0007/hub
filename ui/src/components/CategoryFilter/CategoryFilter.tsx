import React from 'react';
import { useObserver } from 'mobx-react';
import Filter from '../Filter/Filter';

interface store {
  store: any;
}

const CategoryFilter: React.FC<store> = (props: store) => {
  const store = props.store;
  return useObserver(() => <Filter store={store} header="Categories" />);
};

export default CategoryFilter;
