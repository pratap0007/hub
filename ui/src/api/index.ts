import axios from 'axios';
import { API_URL } from '../config/constants';
import { ICategory } from '../store/category';
import { IResource, IVersion } from '../store/resource';

interface Token {
  token: string;
  refreshInterval: string;
  expiresAt: number;
}

interface TokenData {
  access: Token;
  refresh: Token;
}

export interface AuthResponse {
  data: TokenData;
}

export interface Api {
  categories(): Promise<ICategory>;
  resources(): Promise<IResource>;
  resourceVersion(resourceId: number): Promise<IVersion>;
  versionUpdate(versionId: number): Promise<IVersion>;
  authentication(authCode: string): Promise<AuthResponse>;
}

export class Hub implements Api {
  async categories() {
    try {
      return axios.get(`${API_URL}/categories`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resources() {
    try {
      return axios.get(`${API_URL}/resources`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async authentication(authCode: string) {
    try {
      return axios.post(`${API_URL}/auth/login?code=${authCode}`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resourceVersion(resourceId: number) {
    try {
      return axios
        .get(`${API_URL}/resource/${resourceId}/versions`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async versionUpdate(versionId: number) {
    try {
      return axios
        .get(`${API_URL}/resource/version/${versionId}`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }
}
