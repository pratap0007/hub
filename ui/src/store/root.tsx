import React, { ReactChild, ReactChildren } from 'react';
import { types, getEnv, Instance } from 'mobx-state-tree';
import { persist } from 'mst-persist';
import { CategoryStore } from './category';
import { ResourceStore } from './resource';
import { Hub, Api } from '../api';
import { AuthStore } from './auth';

export const Root = types.model('Root', {}).views((self) => ({
  get api(): Api {
    return getEnv(self).api;
  },
  get categories() {
    return getEnv(self).categories;
  },
  get resources() {
    return getEnv(self).resources;
  },
  get user() {
    return getEnv(self).user;
  }
}));

type IRoot = Instance<typeof Root>;

const initRootStore = (api: Api) => {
  const categories = CategoryStore.create({}, { api });
  const resources = ResourceStore.create({}, { api, categories });
  const user = AuthStore.create({ accessTokenInfo: {}, refreshTokenInfo: {} }, { api });

  // This peristently stores user auth details
  persist('authStore', user, {
    storage: localStorage,

    // This adds fields from the store for which data needs to be stored persistently
    whitelist: ['accessTokenInfo', 'refreshTokenInfo', 'isAuthenticated', 'isLoading']
  });

  return Root.create({}, { api, categories, resources, user });
};

interface Props {
  children: ReactChild | ReactChildren;
}

let RootContext: React.Context<IRoot>;
export const useMst = () => React.useContext(RootContext);

export const createProviderAndStore = (api?: Api) => {
  const root = initRootStore(api || new Hub());
  RootContext = React.createContext(root);

  const Provider = ({ children }: Props) => (
    <RootContext.Provider value={root}> {children} </RootContext.Provider>
  );
  return { Provider, root };
};

export const createProvider = (api?: Api) => {
  const { Provider } = createProviderAndStore(api);
  return Provider;
};
