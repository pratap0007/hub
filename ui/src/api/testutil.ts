import { Api } from './';
import * as fs from 'fs';
import { ICategory } from '../store/category';

export class FakeHub implements Api {
  dataDir: string;

  constructor(dataDir: string) {
    this.dataDir = dataDir;
  }

  async categories() {
    const data = `${this.dataDir}/categories.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<ICategory>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }
}
