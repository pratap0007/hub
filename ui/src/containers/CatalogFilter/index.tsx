import React from 'react';
import { useObserver } from 'mobx-react';
import Filter from '../../components/Filter';
import { useMst } from '../../store/root';

const CatalogFilter: React.FC = () => {
  const { catalogs } = useMst();
  return useObserver(() => <Filter store={catalogs} header="Catalog" />);
};

export default CatalogFilter;
