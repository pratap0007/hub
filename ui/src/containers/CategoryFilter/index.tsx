import React from 'react';
import { useObserver } from 'mobx-react';
import { useMst } from '../../store/root';
import Filter from '../../components/Filter';

/* This componnet return Filter component with categories store */
const CategoryFilter: React.FC = () => {
  const { categories } = useMst();
  return useObserver(() => <Filter store={categories} header="Category" />);
};

export default CategoryFilter;
