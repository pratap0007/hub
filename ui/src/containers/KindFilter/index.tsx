import React from 'react';
import { useObserver } from 'mobx-react';
import Filter from '../../components/Filter';
import { useMst } from '../../store/root';

/* This componnet return Filter component with kinds store */
const KindFilter: React.FC = () => {
  const { resources } = useMst();
  return useObserver(() => <Filter store={resources.kinds} header="Kind" />);
};

export default KindFilter;
