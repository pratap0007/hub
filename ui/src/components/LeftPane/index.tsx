import React from 'react';
import { useObserver } from 'mobx-react';
import { GridItem, Grid } from '@patternfly/react-core';
import CatalogFilter from '../../containers/CatalogFilter';
import KindFilter from '../../containers/KindFilter';
import CategoryFilter from '../../containers/CategoryFilter';
import Sort from '../../containers/SortDropDown';
import './LeftPane.css';

const LeftPane: React.FC = () => {
  return useObserver(() => (
    <Grid hasGutter className="hub-leftpane">
      <GridItem span={8}>
        <Sort />
      </GridItem>
      <GridItem>
        <KindFilter />
      </GridItem>
      <GridItem>
        <CatalogFilter />
      </GridItem>
      <GridItem>
        <CategoryFilter />
      </GridItem>
    </Grid>
  ));
};
export default LeftPane;
