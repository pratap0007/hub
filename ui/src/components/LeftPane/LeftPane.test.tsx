import React from 'react';
import LeftPane from '.';
import { mount } from 'enzyme';

import KindFilter from '../../containers/KindFilter';
import CatalogFilter from '../../containers/CatalogFilter';
import CategoryFilter from '../../containers/CategoryFilter';
import PlatformFilter from '../../containers/PlatformFilter';
import { FakeHub } from '../../api/testutil';
import { createProviderAndStore } from '../../store/root';
const TESTDATA_DIR = `src/store/testdata`;
const api = new FakeHub(TESTDATA_DIR);
const { Provider } = createProviderAndStore(api);

describe('LeftPane', () => {
  it('should find the components and match the count', () => {
    const component = mount(
      <Provider>
        <LeftPane />
      </Provider>
    );

    expect(component.debug()).toMatchSnapshot();

    expect(component.find(KindFilter).length).toEqual(1);
    expect(component.find(CatalogFilter).length).toEqual(1);
    expect(component.find(PlatformFilter).length).toEqual(1);
    expect(component.find(CategoryFilter).length).toEqual(1);
  });
});
