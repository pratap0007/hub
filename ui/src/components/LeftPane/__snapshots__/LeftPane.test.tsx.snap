// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftPane should find the components and match the count 1`] = `
"<Grid hasGutter={true} className=\\"hub-leftpane\\">
  <GridItem span={8}>
    <Sort />
  </GridItem>
  <GridItem>
    <KindFilter />
  </GridItem>
  <GridItem>
    <CatalogFilter />
  </GridItem>
  <GridItem>
    <CategoryFilter />
  </GridItem>
</Grid>"
`;
